src/cli/self_update/windows.rs

```diff
@@ -285,6 +282,75 @@ pub fn do_remove_from_path() -> Result<()> {
     _apply_new_path(new_path)
 }
 
+pub fn do_add_to_programs() -> Result<()> {
+    use std::path::PathBuf;
+
+    let key = RegKey::predef(HKEY_CURRENT_USER)
+        .create_subkey(r"Software\Microsoft\Windows\CurrentVersion\Uninstall\rustup")
+        .chain_err(|| ErrorKind::PermissionDenied)?
+        .0;
+
+    // Don't overwrite registry if rustup is already installed
+    let prev = key
+        .get_raw_value("UninstallString")
+        .map(|val| from_winreg_value(&val));
+    if let Ok(Some(s)) = prev {
+        let mut path = PathBuf::from(OsString::from_wide(&s));
+        path.pop();
+        if path.exists() {
+            return Ok(());
+        }
+    }
+
+    let mut path = utils::cargo_home()?;
+    path.push("bin\rustup.exe");
+    let mut uninstall_cmd = path.into_os_string();
+    uninstall_cmd.push(" self uninstall");
```

Comment:
This doesn't seem to correctly quote the path if the home directory contains spaces. I'm not sure if we can use a REG_MULTI_SZ here? That would be better.

Reply:
REG_MULTI_SZ only use the first string as if it was a REG_SZ, so it doesn't help here. Quoting the path should work.